go/plugins/googleai/googleai.go

```diff
@@ -30,79 +30,75 @@ import (
 
 const provider = "google-genai"
 
-// Config provides configuration options for the Init function.
-type Config struct {
-	// API key. Required.
-	APIKey string
-	// Generative models to provide.
-	// If empty, a complete list will be obtained from the service.
-	Models []string
-	// Embedding models to provide.
-	// If empty, a complete list will be obtained from the service.
-	Embedders []string
+var state struct {
```

Comment:
The bigger problem is the illusion that that would give of being able to create multiple plugin instances of the same kind. Ultimately, the registry is global and it's a constraint that its keys must be unique. You couldn't write
```
p1 := pinecone.New()
p1. DefineIndexer("foo", ...)
p2 := pineceone.New()
p2.DefineIndexer("foo", ...)
```
The last line will panic, because both keys are "indexer/pinecone/foo".

We could in theory allow the user to pick the second component, like
```
p1 := pinecone.New("my-first-pinecone-plugin")
```
but I think that would diverge too much from the JS approach.

Another way to put it is that the global state in the plugins is just an extension of the global state in the registry.